cmd/system.go

```diff
@@ -43,10 +51,35 @@ var systemCmd = &cobra.Command{
 var systemDeployCmd = &cobra.Command{
 	Use:   "deploy",
 	Short: "Deploy DCE to a new master account",
-	Run: func(cmd *cobra.Command, args []string) {
+	RunE: func(cmd *cobra.Command, args []string) error {
+		// Before running the command, resolve the configuration by using
+		// the default load order.
+		deployConfig.TFInitOptions = *cfg.Coalesce(&deployConfig.TFInitOptions, Config.Terraform.TFInitOptions, stringp(DCETFInitOptionsEnvVar), stringp(Empty))
+		deployConfig.TFApplyOptions = *cfg.Coalesce(&deployConfig.TFApplyOptions, Config.Terraform.TFApplyOptions, stringp(DCETFApplyOptionsEnvVar), stringp(Empty))
+
 		ctx := context.WithValue(context.Background(), constants.DeployConfig, &deployConfig)
 		if err := Service.Deploy(ctx, &deployOverrides); err != nil {
-			log.Fatalln(err)
+			return err
 		}
+		return nil
 	},
+	// If the command was successful, read the config file back in
+	// and if it's different than the running config, then save it.
+	// We don't want to do this unless the command was succesful,
+	// though, because of cases like bad tf opts we don't want to
+	// create an usuable state.
+	PostRunE: func(cmd *cobra.Command, args []string) error {
+		deployConfig.TFInitOptions = *cfg.Coalesce(&deployConfig.TFInitOptions, Config.Terraform.TFInitOptions, stringp(DCETFInitOptionsEnvVar), stringp(Empty))
+		deployConfig.TFApplyOptions = *cfg.Coalesce(&deployConfig.TFApplyOptions, Config.Terraform.TFApplyOptions, stringp(DCETFApplyOptionsEnvVar), stringp(Empty))
+
+		ctx := context.WithValue(context.Background(), constants.DeployConfig, &deployConfig)
+		if err := Service.PostDeploy(ctx); err != nil {
+			return err
+		}
+		return nil
+	},
+}
+
+func stringp(s string) *string {
```

Comment:
man, I wish go would introduce native syntax for this. I see these methods everywhere, it just seems silly

Reply:
Right??? :(